### ISSUE TITLE
Any chance of this being updated for newer love?

### ISSUE THREAD

**Issue by parkerlreed** [Stabyourself/nottetris2#2]
I `love` (heh) this game but love 0.7 has compilation issues with newer gcc/opengl versions. Would be great to be able to play this on current love.

**Comment by Stabyourself** [comment]
The problem is that the box2d lib was updated and changes things drastically, so this would require some additional work. Maybe in the future once a web-version of LÖVE exists.

**Comment by parkerlreed** [comment]
@Stabyourself Thanks for the reply!

**Comment by erpel** [comment]
Now that there actually is a license in there, I'll definitively push my started port to current löve. The biggest problem is, that the old binary is not signed or 32 bit (don't remember which it was) but it does not run on current macOS anymore. I'm actually hesitant to upgrade the last system because I'll loose nottetris2...

**Comment by parkerlreed** [comment]
Linux always welcomes you :)

**Comment by Yajo** [comment]
Yes, although we already lost not tetris 2 in Linux years ago due to old love too... :laughing:

**Comment by parkerlreed** [comment]
It's alive and well over on Arch AUR! (I maintain the packages hehe)

But yeah, would be good to have it all up to date.

**Comment by Yajo** [comment]
Woah really? But did you patch the sources or do you still maintain love 0.7.2?

**Comment by erpel** [comment]
On AUR it looks like the package uses love 0.7: https://aur.archlinux.org/packages/nottetris2/

If someone had ported this to current love, I’d hope they provided patches upstream.

**Comment by parkerlreed** [comment]
The only patch needed for 0.7 is this.

```
diff --unified --recursive --text love-HEAD.orig/src/modules/graphics/opengl/Framebuffer.cpp love-HEAD.new/src/modules/graphics/opengl/Framebuffer.cpp
--- love-HEAD.orig/src/modules/graphics/opengl/Framebuffer.cpp	2019-03-14 12:46:55.032982224 -0400
+++ love-HEAD.new/src/modules/graphics/opengl/Framebuffer.cpp	2019-03-14 12:47:22.356175299 -0400
@@ -1,3 +1,5 @@
+#define GL_GLEXT_PROTOTYPES
+
 #include "Framebuffer.h"
 #include <common/Matrix.h>
 
diff --unified --recursive --text love-HEAD.orig/src/modules/graphics/opengl/SpriteBatch.cpp love-HEAD.new/src/modules/graphics/opengl/SpriteBatch.cpp
--- love-HEAD.orig/src/modules/graphics/opengl/SpriteBatch.cpp	2019-03-14 12:46:55.032982224 -0400
+++ love-HEAD.new/src/modules/graphics/opengl/SpriteBatch.cpp	2019-03-14 12:47:33.346119890 -0400
@@ -18,6 +18,8 @@
 * 3. This notice may not be removed or altered from any source distribution.
 **/
 
+#define GL_GLEXT_PROTOTYPES
+
 #include "SpriteBatch.h"
 
 // STD
```

**Comment by parkerlreed** [comment]
![image](https://user-images.githubusercontent.com/841440/92014387-26434a00-ed1d-11ea-890d-d00d92a23231.png)

**Comment by Yajo** [comment]
However in https://github.com/flathub/flathub/pull/377 @barthalion did a lot more patches and still didn't get to make it work. Are you sure that just that patch would be enough? :thinking:

**Comment by parkerlreed** [comment]
@Yajo That just a patch for love 0.7 to get LOVE itself working on new systems :)

**Comment by Yajo** [comment]
I've added your patch in https://github.com/Yajo/flathub/commit/bb6696d5568e0284e4c411c5fa281581ca4d4328 and managed to build the app following [these instructions](https://docs.flatpak.org/en/latest/first-build.html).

However, it still fails:

```
➤ flatpak run net.stabyourself.nottetris2 
Could not set video mode: X11 driver not configured with OpenGL
Error: [string "boot.lua"]:1: Could not set screen mode
stack traceback:
        [C]: in function 'assert'
        [string "boot.lua"]:1: in function <[string "boot.lua"]:1>
        [C]: in function 'xpcall'
```

Does that tell you anything? (I've never programmed lua or C before)

**Comment by parkerlreed** [comment]
>Could not set video mode: X11 driver not configured with OpenGL
>Could not set screen mode

Maybe the OpenGL passthrough isn't working?

**Comment by parkerlreed** [comment]
If you have Intel/AMD GPU you need the Mesa flatpak to use OpenGL iirc. 

If Nvidia then it's a bit more complicated https://www.linuxuprising.com/2018/06/how-to-get-flatpak-apps-and-games-built.html

**Comment by Yajo** [comment]
> Maybe the OpenGL passthrough isn't working?

Mm no idea of what that means... sorry for being so newbie. How can y check that?

> If Nvidia then it's a bit more complicated https://www.linuxuprising.com/2018/06/how-to-get-flatpak-apps-and-games-built.html

Done, but it doesn't work either:

```
➤ rpm -q kmod-nvidia
kmod-nvidia-455.28-1.fc32.x86_64

➤ flatpak install flathub org.freedesktop.Platform.GL32.nvidia-455-28 org.freedesktop.Platform.GL.nvidia-455-28
Looking for matches…
Skipping: org.freedesktop.Platform.GL.nvidia-455-28/x86_64/1.4 is already installed


        ID                                                      Branch             Op            Remote             Descarga
 1. [✓] org.freedesktop.Platform.GL32.nvidia-455-28             1.4                i             flathub            352,5 kB / 169,4 MB

Installation complete.

➤ flatpak-builder build-dir --force-clean net.stabyourself.nottetris2.yml
Emptying app dir 'build-dir'
Downloading sources
Starting build of net.stabyourself.nottetris2
Cache hit for SDL1, skipping build
Cache hit for glu, skipping build
Cache hit for lua, skipping build
Cache hit for libpng, skipping build
Cache hit for libmng, skipping build
Cache hit for physfs, skipping build
Cache hit for libdevil, skipping build
Cache hit for libmodplug, skipping build
Cache hit for love, skipping build
Cache hit for nottetris2, skipping build
Cache hit for cleanup, skipping
Cache hit for finish, skipping
Everything cached, checking out from cache
Pruning cache

➤ flatpak-builder --user --install --force-clean build-dir net.stabyourself.nottetris2.yml
Emptying app dir 'build-dir'
Downloading sources
Starting build of net.stabyourself.nottetris2
Cache hit for SDL1, skipping build
Cache hit for glu, skipping build
Cache hit for lua, skipping build
Cache hit for libpng, skipping build
Cache hit for libmng, skipping build
Cache hit for physfs, skipping build
Cache hit for libdevil, skipping build
Cache hit for libmodplug, skipping build
Cache hit for love, skipping build
Cache hit for nottetris2, skipping build
Cache hit for cleanup, skipping
Cache hit for finish, skipping
Everything cached, checking out from cache
Exporting net.stabyourself.nottetris2 to repo
Commit: f8cc85d11f29f63175846a6f0b0a3155fbdaeb7fa06568a3fe152e0bfb287c43
Metadata Total: 89
Metadata Written: 1
Content Total: 103
Content Written: 0
Content Bytes Written: 0 (0 bytes)
Exporting net.stabyourself.nottetris2.Debug to repo
Commit: 032fd3dd1ed48be2e28a873ac3d73916d2bf815f5636429bc7a6f8a155492193
Metadata Total: 241
Metadata Written: 1
Content Total: 935
Content Written: 0
Content Bytes Written: 0 (0 bytes)
Installing app/net.stabyourself.nottetris2/x86_64/master
Pruning cache

➤ flatpak run net.stabyourself.nottetris2
Could not set video mode: X11 driver not configured with OpenGL
Error: [string "boot.lua"]:1: Could not set screen mode
stack traceback:
        [C]: in function 'assert'
        [string "boot.lua"]:1: in function <[string "boot.lua"]:1>
        [C]: in function 'xpcall'
```

**Comment by Yajo** [comment]
I just rebuilt without cache, to see if that was the problem, but it fails equally. Here are the build logs, in case they tell you something: https://gist.github.com/Yajo/4d7d682cbcdf11e7367a8e8ed480a156

**Comment by hollunder** [comment]
Smart as I am I just forked it and started a port to Löve 11.3, figured it wouldn't take long.
I have it starting and the menu is working but I can only agree with the initial assessment that the physics part is the problem. The API changed so much that someone would need to partially rewrite that. I say "someone" because I don't particularly enjoy programming physics.

**Comment by Yajo** [comment]
Did you check https://github.com/Stabyourself/nottetris2/pull/1?

**Comment by SJang1** [comment]
Just a note that this issue can handle the problem for Macs.
As commented before, from the macOS Catalina, x86 apps could not be started, and from late 2020, apple is now changing its architecture to Apple Silicon (ARM) and this would make x64 programs also not able to run anymore in the future. for this reason, it also needs to be available with Apple Silicon Macs... not only x64

**Comment by hollunder** [comment]
I've created a PR from what I have, simply to make it easier to find in case someone wants to continue work on it. https://github.com/Stabyourself/nottetris2/pull/3

**Comment by SJang1** [comment]
For just an update: from now if you build Love from the source Love could run Love natively on apple silicon Mac devices.

**Comment by Cuteistfox** [comment]
Any updates?

**Comment by parkerlreed** [comment]
> Any updates?

Just randomly searched this again and apparently he has an updated repo.

https://github.com/Stabyourself/not-tetris-3

Haven't tested it yet.

**Comment by erpel** [comment]
It's working with love 11.5 but it's not "nottetris2" - It's something with balls instead of bricks. Very much not tetris, and also kind of fun, but very easy at the current state.

**Comment by parkerlreed** [comment]
Ahh thanks for checking! Thought it may have been an actual update, shame.

**Comment by ryandesign** [comment]
> [Not Tetris 3 is] something with balls instead of bricks

@parkerlreed In its current rather unfinished form, Not Tetris 3 seems more like a "not" version of Puyo Puyo than Tetris.

> The biggest problem is, that the old binary is not signed or 32 bit (don't remember which it was) but it does not run on current macOS anymore.

@erpel Signing is not necessary to run an app. The first problem is LÖVE 0.7.x was only released as a 32-bit binary. There are many additional problems after that with getting LÖVE 0.7.x working on modern systems, many of which were fixed in newer branches of LÖVE which of course doesn't help when we want to run a game that requires LÖVE 0.7.x.

I've backported the necessary fixes to get LÖVE 0.7.x (and newer versions) working on modern macOS and have added them to MacPorts, along with Not Tetris 2 and other Stabyourself.net games. Here's [the list of LÖVE games now available in MacPorts](https://ports.macports.org/category/love). If you have any problem with those please let me know by filing a MacPorts bug report.

**Comment by loren-osborn** [comment]
I really know neither Lua nor LÖVE, so I relied on ChatGPT to do most of the heavy lifting, I got gameA mostly working, but am unsure if it's at all a viably playable game? (or am I just lousy at it. )... 

I'm happy to share what I fixed, but it's very half-baked at the moment.

@Stabyourself, do you still have any interest in this game?

**Comment by loren-osborn** [comment]
If anyone's interested, this is what I have... admittedly half-baked:

https://github.com/loren-osborn/nottetris2/tree/losborn_portToLove11.5

**Comment by Stabyourself** [comment]
I'm definitely still interested in the game (purely fun-wise I think Not Tetris 2 is the best game I made)
I'm currently on vacation so I can't really check your work, but things "feeling different" is consistent with my own attempts of just 1:1 porting it to the new apis.

**Comment by loren-osborn** [comment]
I don't have the original to compare things too... ChatGPT is claiming that the default in-game units changed from pixels to meters, so that definitely maps to things feeling different... as a stop gap, instead of finding where the coordinates needed conversion, I temporarily just multiplied forces by 32 and torques by 64... I'm sure fixing things to be in meters proper would help.


As there still seems to be interest, I will, at least, try to clean up the diffs... and maybe add some build tools....

**Comment by loren-osborn** [comment]
I’m continuing to work through this port as a basic, mostly-functional proof-of-concept. I want to clarify upfront that this version is intentionally minimal and rough—particularly, scoring, row detection/removal, and physics object cleanup currently have significant unresolved issues. I’ve temporarily disabled some of these features to bypass problems encountered during the initial port.

My immediate goal is to revert unintended stylistic changes, organize the existing changes clearly, and potentially add a basic Makefile build system. The intent is to deliver a cleaner, structured starting point to make it easier for you to complete the port as you see fit. I just wanted to set clear expectations so you’re prepared for some rough edges at the time of handoff!

**Comment by loren-osborn** [comment]
I will probably add a build mechanism before issuing a PR, but the important changes are in https://github.com/loren-osborn/nottetris2/tree/losborn_portToLove11.5 

Make no mistake... this is still a hot mess.

**Comment by loren-osborn** [comment]
I'm shelving my work on the build system for now. I just don't have time now. The porting work I did is here for review... As I mentioned above, it's still a hot mess: https://github.com/Stabyourself/nottetris2/pull/10

**Comment by loren-osborn** [comment]
As I don't have (and can't currently afford) an Apple developer certificate, I don't anticipate getting code signing fully automated any time soon. I do have the GNU Makefile successfully building a playable macOS `.app` bundle, if this helps anyone.

**Comment by loren-osborn** [comment]
Hi @Stabyourself, I hope you had a fantastic vacation! Just a quick check‑in on two fronts:

1) **LÖVE 11.5 Port**
I’ve **already** finished updating the code so it runs under LÖVE 11.5. My “hot mess” remark was only about a few known crashes and memory leaks I wasn’t confident fixing without your insight into the original architecture. I’d really appreciate your guidance there—and I’d be happy to assist in handing the ball over so we can get it fully across the finish line.

2) **Incomplete Build System**
I’ve put a bit of over‑engineering into a Makefile to automate Mac/Windows/Linux builds. Since the game never had any build tooling, this shouldn't be blocking the port itself. I’m a bit short on time, so I’ll continue developing it once I know the remaining game issues are being addressed—but I’m also happy to give you a quick demo of where it stands if you like. Once it’s all working, I’ll spin it out into its own PR.

No rush—are you back at your desk, or should I check in again later? Thanks again for your interest, and I’m here whenever you’re ready!

**Comment by SJang1** [comment]
> As I don't have (and can't currently afford) an Apple developer certificate, I don't anticipate getting code signing fully automated any time soon. I do have the GNU Makefile successfully building a playable macOS `.app` bundle, if this helps anyone.

Sounds great, I do have Apple Dev Certificate that can sign it, but I am not comfortable to sign others before I understand all the codes. I just want to test out that Makefile, but I can't find Makefile on your repo really fast, could you help me with the Makefile so I would like to test it on my Mac? (which is ported on newer love, like 11.5)

**Comment by loren-osborn** [comment]
Hi @SJang1 —thanks again for your help! A few details:

### 1) Makefile location
I’ve pushed the build system into the `losborn_addBuildSystem` branch:  
https://github.com/loren-osborn/nottetris2/tree/losborn_addBuildSystem  

### 2) Building the macOS bundle today
On your Mac, just run:
```bash
make build/bin/NotTetris2.app
```

That will download LÖVE 11.5, generate the PNG icons from the SVG, create game.love from src/, append it to the LÖVE executable, and assemble build/bin/NotTetris2.app.

### 3) No signing logic yet

Right now the Makefile only produces an unsigned .app. I haven’t added any codesign steps.

### 4) Next step: adding signing

I plan to update the Makefile so it will:
	1.	Detect your Developer ID Application certificate in your macOS keychain
	2.	Run codesign to sign the .app
	3.	Optionally fall back to a self-signed cert if you enable that mode

> [!IMPORTANT]
> No secrets ever live in version control… but it sounds like you already understand that. The Makefile will simply reference your locally installed identity in the keychain at build time.

### 5) Pairing session

I anticipate needing about an hour to implement and test the signing logic together.
I’m in Pacific Daylight Time—what time tomorrow works for you? I can jump on a call or screen-share to get this wired up.

Looking forward to it!

**Comment by SJang1** [comment]
Makefile:186: *** The system is missing the following required tools: convert, rsvg-convert and wine.  Stop.

Looks like that thing which is on  https://github.com/loren-osborn/nottetris2/tree/losborn_addBuildSystem I just downloaded it, doesn't meet the Mac packages that's avaliable. (I am working on MacOS now, and seems like that this would be an issue with your Makefile that it seems to may be for only Linux?)

> I anticipate needing about an hour to implement and test the signing logic together.

I am not comfortable being in a call or screen share at least on this week due to my plans, if needed I will jump on in or you could contact on email later for plan adjustment.

**Comment by SJang1** [comment]
@loren-osborn Also just found out that your ported version on MacOS has some problems on removing lines.

```2025-04-23 15:05:51.314 love[93025:16693838] TSM AdjustCapsLockLEDForKeyTransitionHandling - _ISSetPhysicalKeyboardCapsLockLED Inhibit
Error: gameA.lua:507: Box2D assertion failed: m_I > 0.0f
stack traceback:
	[love "boot.lua"]:352: in function <[love "boot.lua"]:348>
	[C]: in function 'destroy'
	gameA.lua:507: in function 'removeline'
	gameA.lua:1019: in function 'checklinedensity'
	gameA.lua:1223: in function 'endblockA'
	gameA.lua:374: in function 'gameA_update'
	main.lua:397: in function 'update'
	[love "callbacks.lua"]:162: in function <[love "callbacks.lua"]:144>
	[C]: in function 'xpcall'
```

**Comment by loren-osborn** [comment]
Hi @SJang1, thanks for trying out the build and for the feedback! I’ll address each point:

---

### 1) Missing tools error

By default the Makefile runs **preflight checks** for all three platforms (macOS, Windows, Linux). That’s why it’s checking for `wine`, `convert`, and `rsvg-convert`.

- **`wine`** is used to run Windows tools when building Windows binaries. It runs on both **macOS and Linux**. If you don’t want to build the Windows version, you can skip it with:

  ```
  make skip_windows build/bin/NotTetris2.app
  ```

- For icon conversion, you only need **one** of the following tools:
  - `convert` (from ImageMagick), **or**
  - `rsvg-convert` (from librsvg)

  These are used to convert the SVG icon to PNG.

- On macOS, you can install one of these tools using a package manager. If you're unfamiliar with these:

  - **[Homebrew](https://brew.sh/)** is a common option. To install, open Terminal and run:
    ```
    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
    ```

    Then install a tool like this:
    ```
    brew install imagemagick
    # or
    brew install librsvg
    ```

  - **MacPorts** is another option, though less commonly used these days:
    ```
    sudo port install imagemagick
    # or
    sudo port install librsvg
    ```

---

### 2) Crash on removing lines

The Box2D assertion failure:

```
Box2D assertion failed: m_I > 0.0f
...
in function 'removeline'
```

is a bug I’ve seen before. I’m currently waiting on guidance from @Stabyourself to fix the underlying physics cleanup. Once we have his input, I can resolve that crash.

---

### 3) Next steps & code signing

- I can continue work on the **code-signing** logic even while you’re busy.  
- When I have it ready, I’d really appreciate your help testing it on your Mac.  
- I know your schedule is full this week, and mine is too, so just let me know when you have time to try it out.

Thanks again for your help, and let me know if you have any questions!

**Comment by SJang1** [comment]
@loren-osborn Thanks for the update. I've found some issues.

Installed imagemagick with Homebrew.

Build warnings:
```
Copied build/bin/NotTetris2.app/Contents/Frameworks and Assembled build/bin/NotTetris2.app/Contents/MacOS/nottetris2
[ -d "build/icons/NotTetris2Icon.iconset" ] || ( mkdir -p "build/icons/NotTetris2Icon.iconset" || (rm -rf "build/icons/NotTetris2Icon.iconset" ; exit 1))
convert -resize 1024x1024 src/graphics/NotTetris2Icon.svg build/icons/NotTetris2Icon.iconset/icon_512x512@2x.png
WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

convert -resize 16x16 src/graphics/NotTetris2Icon.svg build/icons/NotTetris2Icon.iconset/icon_16x16.png
WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

convert -resize 32x32 src/graphics/NotTetris2Icon.svg build/icons/NotTetris2Icon.iconset/icon_16x16@2x.png
WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

cp build/icons/NotTetris2Icon.iconset/icon_16x16@2x.png build/icons/NotTetris2Icon.iconset/icon_32x32.png
convert -resize 64x64 src/graphics/NotTetris2Icon.svg build/icons/NotTetris2Icon.iconset/icon_32x32@2x.png
WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

cp build/icons/NotTetris2Icon.iconset/icon_32x32@2x.png build/icons/NotTetris2Icon.iconset/icon_64x64.png
convert -resize 128x128 src/graphics/NotTetris2Icon.svg build/icons/NotTetris2Icon.iconset/icon_128x128.png
WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

cp build/icons/NotTetris2Icon.iconset/icon_128x128.png build/icons/NotTetris2Icon.iconset/icon_64x64@2x.png
convert -resize 256x256 src/graphics/NotTetris2Icon.svg build/icons/NotTetris2Icon.iconset/icon_128x128@2x.png
WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

cp build/icons/NotTetris2Icon.iconset/icon_128x128@2x.png build/icons/NotTetris2Icon.iconset/icon_256x256.png
convert -resize 512x512 src/graphics/NotTetris2Icon.svg build/icons/NotTetris2Icon.iconset/icon_256x256@2x.png
WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

cp build/icons/NotTetris2Icon.iconset/icon_256x256@2x.png build/icons/NotTetris2Icon.iconset/icon_512x512.png
[ -d "build/bin/NotTetris2.app/Contents/Resources" ] || ( mkdir -p "build/bin/NotTetris2.app/Contents/Resources" || (rm -rf "build/bin/NotTetris2.app/Contents/Resources" ; exit 1))
iconutil -c icns build/icons/NotTetris2Icon.iconset -o build/bin/NotTetris2.app/Contents/Resources/NotTetris2Icon.icns
exit 1 ; # need to sign app bundle
make: *** [build/bin/NotTetris2.app] Error 1
rm build/icons/.make: unlink: build/icons/.: Operation not permitted

```

If librsvg also installed, no big issue on log
```
[ -d "build/bin/NotTetris2.app/Contents/Resources" ] || ( mkdir -p "build/bin/NotTetris2.app/Contents/Resources" || (rm -rf "build/bin/NotTetris2.app/Contents/Resources" ; exit 1))
iconutil -c icns build/icons/NotTetris2Icon.iconset -o build/bin/NotTetris2.app/Contents/Resources/NotTetris2Icon.icns
exit 1 ; # need to sign app bundle
make: *** [build/bin/NotTetris2.app] Error 1
```

However  I get error when trying to run it:

Run Error:
```
Error: [love "boot.lua"]:303: setIcon only accepts 32-bit RGBA images.
stack traceback:
	[love "boot.lua"]:352: in function <[love "boot.lua"]:348>
	[C]: in function 'setIcon'
	[love "boot.lua"]:303: in function <[love "boot.lua"]:126>
	[C]: in function 'xpcall'
	[love "boot.lua"]:362: in function <[love "boot.lua"]:355>
	[C]: in function 'xpcall'
```

**Comment by loren-osborn** [comment]
Hi @SJang1—thanks again for all your testing! A few follow-ups based on what you found:

---

### 1) **ImageMagick `convert` deprecation warning**  
That one's actually on me—I hadn’t tested the ImageMagick path thoroughly since I had `rsvg-convert` working locally. I’ll make sure the Makefile uses `magick convert` when needed for ImageMagick v7. Thanks for pointing it out!

---

### 2) **"setIcon only accepts 32-bit RGBA images" error**  
That’s a great catch. I haven’t seen that error myself, which makes me wonder if this might be specific to architecture (e.g. Apple Silicon vs Intel Macs), or perhaps differences in how `convert` or `rsvg-convert` encode the PNGs on your setup. I’ll dig into how to guarantee the image is saved in 32-bit RGBA format, and might switch the icon generation to default to `rsvg-convert` where available.

---

### 3) **`unlink` error on `build/icons/.`**  
That’s a side effect of how my Makefile uses directory targets. `make` tries to remove `.`, which isn’t allowed and isn’t a big deal—it’s just cleanup noise. I’m aware of it and not concerned for now, but thanks for flagging it just in case.

---

### 4) **`exit 1 ; # need to sign app bundle`**  
That line is intentional—just a placeholder until I finish adding the code-signing logic. Once that’s in place, the bundle should sign automatically if a valid certificate is found.

---

This feedback has been super helpful—really appreciate the time you’re putting into this! Let me know if anything else pops up.

**Comment by loren-osborn** [comment]
Just a thought... try doing a

```
make skip_windows clean ; make skip_windows build/bin/NotTetris2.app
```

To see if that gets the 32-bit error to go away... it might be because you originally built the icons with ImageMagick.

**Comment by loren-osborn** [comment]
I’m really glad I had the chance to see a real user interact with the Makefile—it definitely revealed a few blind spots I was too close to catch. The logic for computing the desired default targets is mostly in place, but since many of those targets haven’t been implemented yet, it doesn’t amount to working behavior just yet. I also overlooked a few edge cases around how explicit targets are handled.

Because of that, I’m going to hold off on adding the code signing logic while I focus on cleaning up target handling and getting more of the default targets into working shape. I’ve got some higher-priority items on my plate at the moment, so realistically this part may take at least a week.

I’ll keep you posted here as I make progress.

**Comment by loren-osborn** [comment]
I just noticed and fixed the name of my PR (#10) which was very poorly named.

@Stabyourself, Have you had a chance to review any of my changes?

@SJang1, I'm still working on the makefile, but I got distracted by family commitments. I'll keep you updated.

**Comment by Stabyourself** [comment]
Hey, I'll get to looking at this as soon as I find the time.
